Seymour/Matroid/Graphicness.lean

```diff
@@ -175,3 +175,115 @@ theorem Matroid.IsGraphic.isRegular {M : Matroid α} (hM : M.IsGraphic) :
     M.IsRegular := by
   obtain ⟨X, Y, A, hA, hMA⟩ := hM
   exact ⟨X, Y, A, hA.isTotallyUnimodular, hMA⟩
+
+
+open scoped BigOperators Matrix
+
+variable {X Y E ι₁ ι₂ 𝔽 : Type*}
+variable [Field 𝔽]
+
+/-- Dot product on `E → 𝔽` using a finite sum. -/
+def dot [Fintype E] (v w : E → 𝔽) : 𝔽 :=
+  ∑ e, v e * w e
+
+def rowSpace {ι E 𝕂 : Type*} [Field 𝕂] (A : Matrix ι E 𝕂) : Submodule 𝕂 (E → 𝕂) :=
+  Submodule.span 𝕂 (Set.range (fun i => A i))
+
+/-- Orthogonal complement (with respect to `dot`) of a submodule of `E → 𝔽`. -/
+def orth [Fintype E] (U : Submodule 𝔽 (E → 𝔽)) : Submodule 𝔽 (E → 𝔽) :=
+{ carrier := { v | ∀ u, u ∈ U → dot (E := E) v u = 0 }
+  zero_mem' := by
+    intro u hu
+    simp [dot]
+  add_mem' := by
+    intro v w hv hw u hu
+    sorry
+  smul_mem' := by
+    intro a v hv u hu
+    sorry }
+
+/-- Standard block matrix `A = [1_X | B]` with columns indexed by `Sum X Y`. -/
+def stdMat [DecidableEq X] (B : Matrix X Y 𝔽) : Matrix X (Sum X Y) 𝔽 :=
+  fun x e =>
+    match e with
+    | Sum.inl x' => if x = x' then 1 else 0
+    | Sum.inr y  => B x y
+
+/-- The "dual" block matrix `B* = -Bᵀ` over an arbitrary field. -/
+def dualBlock (B : Matrix X Y 𝔽) : Matrix Y X 𝔽 :=
```

Comment:
Such a small thing is better to inline than to have as a definition that you would have to unfold all the time.